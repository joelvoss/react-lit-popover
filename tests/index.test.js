import * as React from 'react';
import { render } from './test-utils';
import { axe } from 'jest-axe';

import {
	Popover,
	positionDefault,
	positionMatchWidth,
	positionRight,
} from '../src/index';

describe('<Popover />', () => {
	it('should not have ARIA violations', async () => {
		const Element = () => {
			const ref = React.useRef(null);
			return (
				<div>
					<div ref={ref} />
					<Popover targetRef={ref}>
						<div>I'm inside a popover!</div>
					</Popover>
				</div>
			);
		};

		let { container } = render(<Element />);
		expect(await axe(container)).toHaveNoViolations();
	});

	it(`should render a popover`, async () => {
		const Element = () => {
			const ref = React.useRef(null);
			return (
				<div>
					<div ref={ref} />
					<Popover targetRef={ref}>
						<div>I'm inside a popover!</div>
					</Popover>
				</div>
			);
		};

		const { baseElement } = render(<Element />);
		expect(baseElement).toMatchSnapshot();
	});

	it(`should render a popover with a different position`, async () => {
		const Element = () => {
			const ref = React.useRef(null);
			return (
				<div>
					<div ref={ref} />
					<Popover targetRef={ref} position={positionMatchWidth}>
						<div>I'm inside a popover!</div>
					</Popover>
				</div>
			);
		};

		const { baseElement } = render(<Element />);
		expect(baseElement).toMatchSnapshot();
	});
});

describe('positionDefault', () => {
	it('should calculate the correct css position', () => {
		const targetRect = {
			bottom: 390,
			height: 17,
			left: 1065,
			right: 1170,
			top: 373,
			width: 105,
			x: 1065,
			y: 373,
		};

		const popoverRect = {
			bottom: 230,
			height: 24,
			left: 1065,
			right: 1385,
			top: 206,
			width: 320,
			x: 1065,
			y: 206,
		};

		expect(positionDefault(targetRect, popoverRect)).toEqual({
			left: '850px',
			top: '390px',
		});
	});
});

describe('positionMatchWidth', () => {
	it('should calculate the correct css position', () => {
		const targetRect = {
			bottom: 390,
			height: 17,
			left: 1065,
			right: 1170,
			top: 373,
			width: 105,
			x: 1065,
			y: 373,
		};

		const popoverRect = {
			bottom: 230,
			height: 24,
			left: 1065,
			right: 1385,
			top: 206,
			width: 320,
			x: 1065,
			y: 206,
		};

		expect(positionMatchWidth(targetRect, popoverRect)).toEqual({
			left: '1065px',
			top: '390px',
			width: '105px',
		});
	});
});

describe('positionRight', () => {
	it('should calculate the correct css position', () => {
		const targetRect = {
			bottom: 390,
			height: 17,
			left: 1065,
			right: 1170,
			top: 373,
			width: 105,
			x: 1065,
			y: 373,
		};

		const popoverRect = {
			bottom: 230,
			height: 24,
			left: 1065,
			right: 1385,
			top: 206,
			width: 320,
			x: 1065,
			y: 206,
		};

		expect(positionRight(targetRect, popoverRect)).toEqual({
			left: '850px',
			top: '390px',
		});
	});
});
