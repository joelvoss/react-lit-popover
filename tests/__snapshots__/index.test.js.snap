// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> should render a popover 1`] = `
<body>
  <div>
    <div>
      <div />
    </div>
  </div>
  <portal>
    <div
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div>
        I'm inside a popover!
      </div>
    </div>
  </portal>
</body>
`;

exports[`<Popover /> should render a popover with a different position 1`] = `
<body>
  <div>
    <div>
      <div />
    </div>
  </div>
  <portal>
    <div
      style="position: absolute; width: 0px; left: 0px; top: 0px;"
    >
      <div>
        I'm inside a popover!
      </div>
    </div>
  </portal>
</body>
`;
